doc/source/dev/plugins.rst

```diff
@@ -0,0 +1,88 @@
+.. devguide_plugins:
+
+Developing Plugins
+------------------
+
+As discussed in the user's guide, WATTS has a variety of plugins for different
+codes (primarily focused on application codes in nuclear science and
+engineering) that are already available. If you wish to use WATTS with a code
+that does not already have a plugin available, extending WATTS to handle another
+code can be done quite easily.
+
+There are two ways to incorporate new codes into WATTS. First, if the code uses
+a text-based input, the :class:`~watts.PluginTemplate` class can be used to
+specify the executable, how it should be invoked from a command-line, and a
+template for the input file. For example, if you have a code called ``goblin``
+that should be executed from a command line as:
+
+.. code-block:: sh
+
+    goblin input=filein.gbl
```

Comment:
what if the execution is 'goblin -i fill-in.gbl' ?

Reply:
I think this is responded by my question on `execute_command`

Reply:
In that case, you'd pass `execute_command=['{self.executable}', '-i', '{self.input_name}']`. Or, I'm going to give an option such that you could pass it as `execute_command='{self.executable} -i {self.input_name}'`